prereise/gather/demanddata/nrel_efs/get_efs_data.py

```diff
@@ -288,6 +392,259 @@ def partition_by_sector(es, ta, year, sect=None, fpath="", save=False):
+    :return: (*dict*) -- A dict of pandas.DataFrame objects that contain flexibility
+        data for each state and time step in the specified sectors.
+    :raises TypeError: if save is not input as a bool.
+    """
+
+    # Account for the immutable default parameters
+    if sect is None:
+        sect = {"All"}
+
+    # Check the inputs
+    es = _check_electrification_scenarios_for_partition(es)
+    ta = _check_technology_advancements_for_partition(ta)
+    flex = _check_flexibility_scenario(flex)
+    _check_year(year)
+    sect = _check_sectors(sect)
+    fpath = _check_path(fpath)
+    if not isinstance(save, bool):
+        raise TypeError("save must be input as a bool.")
+
+    # Specify the file name and path
+    csv_name = "EFSFlexLoadProfiles_" + es + ".csv"
+    csv_path = os.path.join(fpath, csv_name)
+
+    # Download the specified NREL EFS dataset if it is not already downloaded
+    if not os.path.isfile(csv_path):
+        download_flexibility_data({es}, fpath)
+
+    # Load the data from the downloaded .csv file as a DataFrame
+    df = pd.read_csv(csv_path)
+
+    # Trim the DataFrame for only the specified year
+    df = df.loc[df["Year"] == year]
+
+    # Trim the DataFrame for only the specified technology advancement
+    df = df.loc[df["TechnologyAdvancement"] == ta]
+
+    # Trim the DataFrame for only the specified flexibility scenario
+    df = df.loc[df["Flexibility"] == flex]
+
+    # Drop unnecessary "Year", "Electrification", and "TechnologyAdvancement" columns
+    df.drop(
+        columns=["Year", "Electrification", "TechnologyAdvancement", "Flexibility"],
+        inplace=True,
+    )
+
+    # Split the flexibility DataFrame by sector
+    sect_flex = {
+        i: df[df["Sector"] == i]
+        .drop(columns=["Sector"])
+        .groupby(["LocalHourID", "State"])
+        .sum()
+        .unstack()
+        for i in sect
+    }
+    sect_flex = {
+        i: sect_flex[i].set_axis(sorted(set(abv2state) - {"AK", "HI"}), axis="columns")
+        for i in sect
+    }
+
+    # Add extra day's worth of flexibility to account for leap year
+    sect_flex = {i: account_for_leap_year(sect_flex[i]) for i in sect}
+
+    # Include the appropriate timestamps for the local time (with year=2016)
+    sect_flex = {
+        i: sect_flex[i].set_axis(
+            pd.date_range("2016-01-01", "2017-01-01", freq="H", closed="left"),
+            axis="index",
+        )
+        for i in sect
+    }
+    sect_flex = {i: sect_flex[i].rename_axis("Local Time", axis="index") for i in sect}
+
+    # Save the sectoral DataFrames to .csv files, if desired
+    if save:
+        for i in sect:
+            new_csv_name = (
+                i
+                + "_"
+                + flex
+                + "_Flexibility_"
+                + es
+                + "_"
+                + ta
+                + "_"
+                + str(year)
+                + ".csv"
+            )
```

Comment:
`new_csv_name = f"{i}_{flex}_Flexibility_{es}_{ta}_{year}.csv"`

Reply:
Yeah, this is much cleaner! I just finished updating all file names to use the f-strings, as suggested above.

Reply:
Done.